pkg/odo/cli/init/init.go

```diff
@@ -35,7 +35,22 @@ const RecommendedCommandName = "init"
 var initExample = templates.Examples(`
   # Boostrap a new component in interactive mode
   %[1]s
-`)
+
+  # Bootstrap a new component with a specific devfile from registry
+  %[1]s --name my-app --devfile nodejs
+  
+  # Bootstrap a new component with a specific devfile from a specific registry
+  %[1]s --name my-app --devfile nodejs --devfile-registry MyRegistry
+  
+  # Bootstrap a new component with a specific devfile from the local filesystem
+  %[1]s --name my-app --devfile-path $HOME/devfiles/nodejs/devfile.yaml
+  
+  # Bootstrap a new component with a specific devfile from the web
+  %[1]s --name my-app --devfile-path https://devfiles.example.com/nodejs/devfile.yaml
```

Comment:
Not directly related to the changes in this PR, but a question popped up in my head. Is it possible to preserve the starter project provided by devfile, a la #5318, when using `odo init`? Should it be possible? If yes, is it relevant for v3alpha1? IMO, no. The command I am talking about is:

```sh
$ odo init --name my-app --devfile-path https://devfiles.example.com/nodejs/devfile.yaml --starter nodejs-starter
```

Reply:
Yes, the implementation takes care of the case #5318.

https://github.com/redhat-developer/odo/blob/main/pkg/odo/cli/init/init.go#L158-L180